src/services/repos/useRepos.tsx

```diff
@@ -0,0 +1,271 @@
+  })
+  .nullable()
+
+export type RepositoryResult = z.infer<typeof RepositorySchema>
+
+const RequestSchema = z.object({
+  owner: z
+    .object({
+      repositories: z
+        .object({
+          edges: z.array(
+            z.object({
+              node: RepositorySchema,
+            })
+          ),
+          pageInfo: z.object({
+            hasNextPage: z.boolean(),
+            endCursor: z.string().nullable(),
+          }),
+        })
+        .nullable(),
+    })
+    .nullable(),
+})
+
+const RequestSchemaMyRepos = z.object({
+  me: z
+    .object({
+      viewableRepositories: z
+        .object({
+          edges: z.array(
+            z.object({
+              node: RepositorySchema,
+            })
+          ),
+          pageInfo: z.object({
+            hasNextPage: z.boolean(),
+            endCursor: z.string().nullable(),
+          }),
+        })
+        .nullable(),
+    })
+    .nullable(),
+})
+
+function fetchMyRepos({
+  provider,
+  variables,
+  after,
+  signal,
+}: FetchMyReposArgs) {
+  const query = `
+    query MyRepos($filters: RepositorySetFilters!, $ordering: RepositoryOrdering!, $direction: OrderingDirection!, $after: String) {
+        me {
+          viewableRepositories(filters: $filters, ordering: $ordering, orderingDirection: $direction, first: $first, after: $after) {
+            edges {
+              node {
+                ...RepoForList
+              }
+            }
+            pageInfo {
+                hasNextPage
+                endCursor
+            }
+          }
+        }
+      }
+
+      ${repositoryFragment}
+  `
+
+  return Api.graphql({
+    provider,
+    query,
+    signal,
+    variables: { ...variables, after },
+  }).then((res) => {
+    const parsedRes = RequestSchemaMyRepos.safeParse(res?.data)
+
+    if (!parsedRes.success) {
+      return Promise.reject({
+        status: 404,
+        data: null,
+      })
+    }
+
+    const me = parsedRes.data.me
+    return {
+      repos: mapEdges(me?.viewableRepositories),
+      pageInfo: me?.viewableRepositories?.pageInfo,
+    }
+  })
+}
+
+interface FetchReposForOwnerArgs {
+  provider: string
+  variables: RepoQueryVariables
+  owner: string
+  after: string
+  signal?: AbortSignal
+}
+
+function fetchReposForOwner({
+  provider,
+  variables,
+  owner,
+  after,
+  signal,
+}: FetchReposForOwnerArgs) {
+  const query = `
+    query ReposForOwner($filters: RepositorySetFilters!, $owner: String!, $ordering: RepositoryOrdering!, $direction: OrderingDirection!, $after: String, $first: Int) {
+        owner(username: $owner) {
+          repositories(filters: $filters, ordering: $ordering, orderingDirection: $direction, first: $first, after: $after) {
+            edges {
+              node {
+                ...RepoForList
+              }
+            }
+            pageInfo {
+                hasNextPage
+                endCursor
+            }
+          }
+        }
+      }
+
+      ${repositoryFragment}
+  `
```

Comment:
`l`: This is me being just a little picky, but can you update these with their formatted versions so they're easier to read:

```graphql
query ReposForOwner(
  $filters: RepositorySetFilters!
  $owner: String!
  $ordering: RepositoryOrdering!
  $direction: OrderingDirection!
  $after: String
  $first: Int
) {
  owner(username: $owner) {
    repositories(
      filters: $filters
      ordering: $ordering
      orderingDirection: $direction
      first: $first
      after: $after
    ) {
      edges {
        node {
          ...RepoForList
        }
      }
      pageInfo {
        hasNextPage
        endCursor
      }
    }
  }
}

${repositoryFragment}
```